docs/modules/ROOT/pages/advanced-how-tos/using-trusted-artifacts.adoc

```diff
@@ -0,0 +1,38 @@
+= Using Trusted Artifacts
+
+Konflux relies on Trusted Artifacts to securely allow sharing of data between Tasks without
+compromising the build process. See
+link:https://konflux-ci.dev/architecture/ADR/0036-trusted-artifacts.html[ADR-36] for more
+information.
+
+Today, when a component is onboarded to Konflux, the provided build Pipeline uses Trusted Artifacts.
+However, older components might still be using a version of the build Pipeline which does not use
+Trusted Artifacts. This document explains how to identify if Trusted Artifacts are being used and,
+if not, how to start using them.
+
+== Am I using Trusted Artifacts?
+
+To determine if a certain component is using Trusted Artifacts, inspect its build Pipeline. Look for
+the PipelineRun YAML files under the `.tekton` directory in the git repository of the component. If
+the PipelineRun contains a workspace named `workspace`, it is likely NOT using Trusted Artifacts.
+
+== Migrate to Trusted Artifacts
+
+For components already onboarded to Konflux, the simplest way to start using Trusted Artifacts is
+via the link:https://github.com/zregvart/build-migration[build-migration] tool. Run the
+`migration.py` script with the path to the `.tekton` directory as the first parameter. For example:
+`python migration.py ~/src/my-component/.tekton`.
+
+If your build Pipeline has been modified in certain ways, e.g. it includes a custom Task, the
+`build-migration` tool may not work as expected. In this case, perform the required modifications
+manually. These are captured in
+link:https://github.com/konflux-ci/build-definitions/blob/main/pipelines/docker-build-oci-ta/patch.yaml[this]
+patch file. Although the patch file cannot be directly applied to your Pipeline, it describes in
+detail all the required changes.
+
+If your build Pipeline includes custom Tasks that require access to the workspace, e.g. it reads the
+component's source code, the Task must be modified to implement the Trusted Artifacts pattern. At a
+high-level this means accessing data created by other Tasks via OCI artifacts instead of a shared
+workspace backed by a PVC (Persistent Volume Claim).
```

Comment:
Should we state that this can be done by adding parameters and the TA stepaction?

Reply:
Hopefully the newly added (annotated) example makes this clear.